kroxylicious-operator/src/main/resources/META-INF/fabric8/kafkaservices.kroxylicious.io-v1.yml

```diff
@@ -37,13 +37,39 @@ spec:
               type: object
             spec:
               type: object
-              required: ["bootstrapServers"]
+              required: [ "bootstrapServers" ]
               properties:
                 bootstrapServers:
                   description: Bootstrap servers of the Kafka cluster.
                   type: string
                   minLength: 1
                   pattern: ^(([^:]+:[0-9]{1,5}),)*([^:]+:[0-9]{1,5})$
+                nodeIdRanges:
+                  type: array
+                  minItems: 1
+                  description: |
+                    Declares the ranges of node ids present in the Kafka Cluster this KafkaService refers to. Default 
+                    value is a single range starting with node id 0 and ending with 2 inclusive. Note that if you wish
+                    to expose this cluster using plain (non-TLS) connections it is desirable to use a compact set of
+                    node id ranges, as every node id in every range requires a distinct listening port.
+                    See also: https://kafka.apache.org/documentation/#brokerconfigs_node.id
+                  items:
+                    type: object
+                    required: [ "name", "start", "end" ]
+                    x-kubernetes-validations:
+                      - rule: "self.start <= self.end"
+                        message: "end should be greater than or equal to start"
+                    properties:
+                      name:
```

Comment:
name is required and must be unique in the proxy config, we could default it to "range-${index}" so users do not have to supply it, but it's available for customization?

Reply:
answering my own question, I can imagine some users or programmatically built CRs not wanting to care about setting up a unique name, lets set up a default.